### ISSUE TITLE
Error messages are not descriptive

### ISSUE THREAD

**Issue by alexpiet** [AllenNeuralDynamics/dynamic-foraging-task#43]
Error messages printed into the system out are not informative, for example "An error occured: could not convert string to float"

We should at least print the file and line number of the error for debugging

**Comment by alexpiet** [comment]
Looking through the code, it appears there are a lot of try/except blocks where the except is just "pass". This concerns me that we have a lot of silently failing functions. We should aim to be verbose and fail loudly.

**Comment by alexpiet** [comment]
I've taken a first step of updating lines like:
```
try:
    <code>
except:
    pass
```
to
```
try:
    <code>
except Exception as e:
    print('<filename>.<function name>: ',str(e))
```

Should we have a more formal system of logging errors? @bruno-f-cruz @hanhou @XX-Yin

**Comment by bruno-f-cruz** [comment]
The one thing I am a bit picky is to try not to use a catch for all Exceptions and instead have the type of exception you expect to crash the clause. Eg: 

x=1
y="a"
try: 
    z = x + y 
except TypeError: 
    Handle the TypeError exception here

**Comment by alexpiet** [comment]
Yes @bruno-f-cruz I completely agree we should catch specific errors. In this case I actually think many of the exceptions could be addressed by if/else statements.

I'm thinking a little bigger picture, in the event that an exception happens that is not caught by specific error catching, how should we log it?

**Comment by alexpiet** [comment]
@hanhou @XX-Yin and I discussed this issue. We decided that we should create an error log for the python GUI. We decided this log should operate on the scope of one log file for each time the GUI is started. That means that a computer that runs multiple GUIs (for multiple rigs) will have multiple log files. It means that if the GUI is started and multiple sessions are run one after another then all errors from all of those rigs will be in the same log file. 

- Each error should include the computer, rig, session, time, file/function that generated it, and a descriptive message. 
- The process that pulls data from the local computers to the cloud can also pull this file. 
- Our goal should be to log unexpected behavior, and then we can resolve these issues
